ament_cmake_cppcheck/cmake/ament_cmake_cppcheck_lint_hook.cmake

```diff
@@ -24,5 +26,30 @@ file(GLOB_RECURSE _source_files FOLLOW_SYMLINKS
 )
 if(_source_files)
   message(STATUS "Added test 'cppcheck' to perform static code analysis on C / C++ code")
-  ament_cppcheck()
+
+  # Get include paths for added targets
+  set(_all_include_dirs "")
+  # BUILDSYSTEM_TARGETS only supported in CMake >= 3.7
+  if(NOT CMAKE_VERSION VERSION_LESS "3.7.0")
+    get_directory_property(_build_targets DIRECTORY ${CMAKE_SOURCE_DIR} BUILDSYSTEM_TARGETS)
+    foreach(_target ${_build_targets})
+      get_property(_include_dirs
+        TARGET ${_target}
+        PROPERTY INCLUDE_DIRECTORIES
+      )
+
+      # Only append include directories that are from the package being tested
+      # This accomplishes two things:
+      #     1. Reduces execution time (less include directories to search)
+      #     2. cppcheck will not check for errors in external packages
```

Comment:
This fails on Windows in this case: ros2/ros2#942.

Reply:
I don't understand the case that's failing. From the referenced ticket, it looks like the failures are due to headers from external packages not being passed to cppcheck (which is the intended behavior). For this case, we have `ament_cmake_cppcheck_ADDITIONAL_INCLUDE_DIRS` that can be set (introduced in #125).

Reply:
Without the proper include directories `cppcheck` warns for unknown macros.

ros2/rclpy#577 sets an additional include dir.